src/platform/tls_openssl.c

```diff
@@ -249,10 +253,48 @@ CxPlatTlsCertificateVerifyCallback(
         return FALSE;
     }
 
+    if (TlsContext->SecConfig->Flags & QUIC_CREDENTIAL_FLAGS_USE_PORTABLE_CERTIFICATES) {
+        if (Cert) {
+            PortableCertificate.Length = i2d_X509(Cert, &PortableCertificate.Buffer);
+            if (!PortableCertificate.Buffer) {
+                QuicTraceEvent(
+                    TlsError,
+                    "[ tls][%p] ERROR, %s.",
+                    TlsContext->Connection,
+                    "Failed to serialize certificate context");
+                return FALSE;
+            }
+        }
+        if (x509_ctx) {
+            int ChainCount;
+            STACK_OF(X509)* Chain = X509_STORE_CTX_get0_chain(x509_ctx);
+            if ((ChainCount = sk_X509_num(Chain)) > 0) {
+                PKCS7* p7 = PKCS7_new();
```

Comment:
Is PKCS7 more commonly used in these scenarios, and easier to consume than an array of `QUIC_BUFFER` which each represent one X509 certificate in the chain?

Reply:
I don't have any stats. But it is certainly easier to consume. OpenSSL and Windows have already functions for import/export and for .NET this is one line with `X509Certificate2Collection.Import()`. I would hope it would be similar for other languages. 
Array of `QUIC_BUFFER` could possibly work as well but then you have probably more allocations and buffers to manage.

Reply:
Ok, that sounds good to me.